apiclient/harvester_api/models.py

```diff
@@ -70,7 +70,7 @@ class VMSpec:
     # SSH Key
 
     def __init__(self, cpu_cores, memory, description="", reserved_mem="", os_type="",
-                 mgmt_network=True, guest_agent=True):
+                 mgmt_network=True, guest_agent=True, machine_type=None, networks=None):
```

Comment:
If we were to add a network, it wouldn't be able to be the `default` network - is there a way to tell the VM via configuration to "only" use that network and not the other `default` network?

I was thinking that if we just had the ability to set the Networks on the VMSpec directly during initalization of the object we wouldn't need to worry about the juggling of a VM with two NICs.

Since the test was intending to be negative, if the VM only had one NIC and that NIC was tied directly to a VLAN that purposefully didn't have DHCP access.

Reply:
we can update `networks` after the Spec object initialized, the code snippet:
```python
>>> vm = api.vms.Spec(1, 2)
>>> vm.networks
[{'iface': {'model': 'virtio', 'name': 'default', 'masquerade': {}}, 'network': {'name': 'default', 'pod': {}}}]
>>> 
>>> vm.add_network('nic-1', 'the-net-uid')
{'iface': {'model': 'virtio', 'name': 'nic-1', 'bridge': {}}, 'network': {'name': 'nic-1', 'multus': {'networkName': 'the-net-uid'}}}
>>> vm.networks
[{'iface': {'model': 'virtio', 'name': 'default', 'masquerade': {}}, 'network': {'name': 'default', 'pod': {}}}, {'iface': {'model': 'virtio', 'name': 'nic-1', 'bridge': {}}, 'network': {'name': 'nic-1', 'multus': {'networkName': 'the-net-uid'}}}]
>>> vm.mgmt_network = False
>>> vm.networks
[{'iface': {'model': 'virtio', 'name': 'nic-1', 'bridge': {}}, 'network': {'name': 'nic-1', 'multus': {'networkName': 'the-net-uid'}}}]
```

if you want to let the `mgmt-network` not being the first NIC, you can use:
```python
>>> vm = api.vms.Spec(1, 2, mgmt_network=False)
>>> vm.networks
[]
>>> vm.add_network('nic-1', 'the-net-uid')
{'iface': {'model': 'virtio', 'name': 'nic-1', 'bridge': {}}, 'network': {'name': 'nic-1', 'multus': {'networkName': 'the-net-uid'}}}
>>> vm.mgmt_network = True
>>> vm.networks
[{'iface': {'model': 'virtio', 'name': 'nic-1', 'bridge': {}}, 'network': {'name': 'nic-1', 'multus': {'networkName': 'the-net-uid'}}}, {'iface': {'model': 'virtio', 'name': 'default', 'masquerade': {}}, 'network': {'name': 'default', 'pod': {}}}]
```

Reply:
That's a good mention :+1: :smile: 
I've updated it, since I think we can still check for a bogus vlan if it has the mgmt-network being the first vlan and just adding an additional NIC that doesn't have DHCP resolution.